### PR TITLE
ci: improve releasing

### PR DESCRIPTION
**Pull request by kurtisvg**
Does a few things to improve releasing process:
- Moves file to .ci folder
- Updates release script to check for version
- Uses printf instead of "echo" to work better between linux/mac
- removes comments about non-mac (worked on linux for me)

I've also updated the tigger to run on new tags.

### CODE REVIEW

File: .ci/generate_release_table.sh [hunk 1]
```diff
@@ -20,15 +28,13 @@ do
...
 
     # Calculate the SHA256 checksum of the file
-    # Running from mac
     SHA256=$(shasum -a 256 toolbox | awk '{print $1}')
-    # Running from non-mac
-    # SHA256=$(sha256sum toolbox | awk '{print $1}')
 
     # Write the table row
     output_string+="| [$OS/$ARCH]($URL)   | $SHA256 |\n"
+    output_string+="| %-93s | %-65s |\n"
```
Comment (Yuan325) [inline .ci/generate_release_table.sh]: This will create a new row. Is it intentional?

File: .ci/versioned.release.cloudbuild.yaml [hunk 2]
```diff
@@ -124,22 +131,24 @@ steps:
...
-  dynamic_substitutions: true
+  dynamicSubstitutions: true
   logging: CLOUD_LOGGING_ONLY # Necessary for custom service account
 
 substitutions:
   _REGION: us-central1
   _AR_HOSTNAME: ${_REGION}-docker.pkg.dev
   _AR_REPO_NAME: toolbox
   _BUCKET_NAME: genai-toolbox
+  _DOCKER_URI: ${_AR_HOSTNAME}/$PROJECT_ID/${_AR_REPO_NAME}/toolbox
```
Comment (Yuan325) [inline .ci/versioned.release.cloudbuild.yaml]:
```suggestion
  _DOCKER_URI: ${_AR_HOSTNAME}/${PROJECT_ID}/${_AR_REPO_NAME}/toolbox
```

File: .ci/versioned.release.cloudbuild.yaml [hunk 3]
```diff
@@ -12,40 +12,44 @@
...
-    name: 'gcr.io/cloud-builders/docker'
+  - id: "build-docker"
+    name: "gcr.io/cloud-builders/docker"
     waitFor: ['-']
     script: |
         #!/usr/bin/env bash
         export VERSION=$(cat ./cmd/version.txt)
-        docker buildx build -t ${_AR_HOSTNAME}/$PROJECT_ID/${_AR_REPO_NAME}/toolbox:$VERSION \
-            -t ${_AR_HOSTNAME}/$PROJECT_ID/${_AR_REPO_NAME}/toolbox:latest .
+        docker buildx build -t ${_DOCKER_URI}:$REF_NAME -t ${_DOCKER_URI}:latest .
```
Comment (Yuan325) [inline .ci/versioned.release.cloudbuild.yaml]: Tested this locally manually and $REF_NAME doesn't work. error with "invalid reference format"

File: .ci/generate_release_table.sh (hunk 1 above)
Comment (kurtisvg) [reply .ci/generate_release_table.sh]: Somehow I committed and earlier version. Should be fixed now.

File: .ci/versioned.release.cloudbuild.yaml (hunk 3 above)
Comment (kurtisvg) [reply .ci/versioned.release.cloudbuild.yaml]: It's listed here: https://cloud.google.com/build/docs/configuring-builds/substitute-variable-values#using_user-defined_substitutions

It says "When starting a build using gcloud builds submit, you can specify variables that would normally come from triggered builds with the [--substitutions argument](https://cloud.google.com/sdk/gcloud/reference/builds/submit).". Did you try that?

File: .ci/versioned.release.cloudbuild.yaml [hunk 4]
```diff
@@ -55,20 +58,21 @@ steps:
         CGO_ENABLED=0 GOOS=linux GOARCH=amd64 \
           go build -ldflags "-X github.com/googleapis/genai-toolbox/cmd.metadataString=binary.linux.amd64.$VERSION.$(git rev-parse HEAD)" -o toolbox.linux.amd64
```
Comment (Yuan325) [inline .ci/versioned.release.cloudbuild.yaml:59]: This line have the `$(git rev-parse HEAD)` substitution too. Should we replace all these with the $REF_NAME?

File: .ci/versioned.release.cloudbuild.yaml (hunk 3 above)
Comment (Yuan325) [reply .ci/versioned.release.cloudbuild.yaml]: Yeap, tried that. If we want to be able to run from local, we'll have to update the Dockerfile as well. We have this substitution `$(git rev-parse HEAD)`.

**Pull request merged**